### ISSUE TITLE
Update `match` function

### ISSUE THREAD

**Issue by tjbenton** [stylus/stylus#2103]
Currently the [`match`](https://github.com/stylus/stylus/blob/dev/lib/functions/match.js) function only returns a `boolean` which is helpful but It could be more helpful if it where implemented to be like the js match function. It wouldn't break backwards compatibility unless someone was specifically checking for a boolean and not a truthy falsy value.

```js
module.exports = function match(pattern, val) {
  utils.assertType(pattern, 'string', 'pattern');
  utils.assertString(val, 'val');
  var re = new RegExp(pattern.val);
  return val.string.match(re);;
};
``` 

```styl
// the regex to use for the tests
$regex = '^(height|width)?([<>=]{1,})(.*)'

$has-match = 'height>=sm' // will have a match
$no-match = 'lorem ipsum' // won't have a match

$test1 = match($regex, $has-match)
$test2 = match($regex, $no-match)
 
p('$test1: ' $test1)
p('$test2: ' $test2)

if $test1 {
  p('$test1 is truthy')
} else {
  p('$test1 is falsy')
}

if $test2 {
  p('$test2 is truthy')
} else {
  p('$test2 is falsy')
}
```

stdout
```
inspect: '$test1: ' ('height>=sm' 'height' '>=' 'sm')
inspect: '$test2: ' (null)
inspect: '$test1 is truthy'
inspect: '$test2 is falsy'
```

**Comment by tjbenton** [comment]
If you want me to I can submit a pull request to make this update.

**Comment by Panya** [comment]
@tjbenton yep, sure. Feel free to send a PR with these changes (it would be cool if you add some docs for this function :))

**Comment by Panya** [comment]
Closed via #2109
